sigtrap_posix.go

```diff
@@ -76,6 +76,9 @@ func trapSignalsPosix() {
 					caddyfileToUse = newCaddyfile
 				}
 
+				// Purge the old event hooks
+				purgeEventHooks()
```

Comment:
the potential drawback is that we'll lose valid hooks in case of unsuccessful restart

Reply:
@rdrozhdzh I fixed this, thank you!

Reply:
@rdrozhdzh I fixed this, thank you!

Reply:
@rdrozhdzh I fixed this, thank you!!!